sympy/sets/fancysets.py

```diff
@@ -483,6 +492,138 @@ def _intersect(self, other):
             else:
                 return
 
+    def _union(self, other):
+        """
+
+        This function should only be used internally.
+        See Set._union for docstring
+
+        Used within the :class:`Union` class
+
+        Returns
+        =======
+        self._union(other) returns a new, joined ImageSet if self knows how
+        to join itself with other, otherwise it returns ``None``.
+
+        Examples
+        ========
+
+        >>> from sympy import Lambda
+        >>> from sympy.sets.fancysets import ImageSet
+        >>> from sympy import pprint, Union
+        >>> from sympy.core import Dummy, pi, S, Symbol
+        >>> n = Dummy('n')
+        >>> img1 = ImageSet(Lambda(n, 2*n*pi + 3*pi/4), S.Integers)
+        >>> img2 = ImageSet(Lambda(n, 2*n*pi + 7*pi/4), S.Integers)
+        >>> pprint(img1.union(img2), use_unicode=False)
+                3*pi
+        {n*pi + ---- | n in Integers()}
+                 4
+
+        """
+
+        from sympy.polys import Poly
+        from sympy.simplify import simplify
+        from sympy.polys.polyerrors import PolynomialError
+
+        if other is S.EmptySet:
+            return self
+
+        def val_at(imgset, i):
+            base = imgset.base_set
+            if base == S.Reals:
+                return imgset.lamda(i)
+            else:
+                iterable = iter(base)
```

Comment:
@Shekharrajak I understood the logic behind the `_union`. But I cannot understand why using itrable for `base` other than `Reals`, and why not for `Reals`.